sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/functions/FunctionsTest.java

```diff
@@ -1128,6 +1410,207 @@ public void testTruncate() {
+                (1 row)
+                
+                select truncate(-5678.123451::DOUBLE,0);
+                truncate(-5678.123451::DOUBLE,0)
+                -----
+                -5678
+                (1 row)
+                
+                select truncate(-5678.123451::DOUBLE,1);
+                truncate(-5678.123451::DOUBLE,1)
+                -----
+                -5678.1
+                (1 row)
+                
+                select truncate(-5678.123451::DOUBLE,2);
+                truncate(-5678.123451::DOUBLE,2)
+                -----
+                -5678.12
+                (1 row)
+                
+                select truncate(-5678.123451::DOUBLE,3);
+                truncate(-5678.123451::DOUBLE,3)
+                -----
+                -5678.123
+                (1 row)
+                
+                select truncate(-5678.123451::DOUBLE,4);
+                truncate(-5678.123451::DOUBLE,4)
+                -----
+                -5678.1234
+                (1 row)
+                
+                select truncate(-5678.123451::DOUBLE,5);
+                truncate(-5678.123451::DOUBLE,5)
+                -----
+                -5678.12345
+                (1 row)
+                
+                select truncate(-5678.123451::DOUBLE,6);
+                truncate(-5678.123451::DOUBLE,6)
+                -----
+                -5678.123451
+                (1 row)
+                
+                select truncate(-5678.123451::DOUBLE,-1);
+                truncate(-5678.123451::DOUBLE,-1)
+                -----
+                -5670
+                (1 row)
+                
+                select truncate(-5678.123451::DOUBLE,-2);
+                truncate(-5678.123451::DOUBLE,-2)
+                -----
+                -5600
+                (1 row)
+                
+                select truncate(-5678.123451::DOUBLE,-3);
+                truncate(-5678.123451::DOUBLE,-3)
+                -----
+                -5000
+                (1 row)
+                
+                select truncate(-5678.123451::DOUBLE,-4);
+                truncate(-5678.123451::DOUBLE,-4)
+                -----
+                0
+                (1 row)
+                
+                select truncate(5678.123451::DOUBLE,1::tinyint);
+                truncate(5678.123451::DOUBLE,1)
+                -----
+                5678.1
+                (1 row)
+                
+                select truncate(5678.123451::DOUBLE,1::smallint);
+                truncate(5678.123451::DOUBLE,1)
+                -----
+                5678.1
+                (1 row)
+                
+                select truncate(5678.123451::DOUBLE,1::int);
+                truncate(5678.123451::DOUBLE,1)
+                -----
+                5678.1
+                (1 row)
+                
+                """
+        );
+    }
+
+    @Test
+    public void testRoundTruncateEdgeCases() {
+        // Calcite normalizes the sign here
```

Comment:
what is "normalize the sign"?

Reply:
Oh, Calcite returns 0 instead of -0, so we also turn it into 0.0.

Reply:
Called it normalize because I remembered it from: https://docs.rs/rust_decimal/1.34.2/rust_decimal/struct.Decimal.html#method.normalize